### ISSUE TITLE
[Bug]: querynode memory is not balance in k8s

### ISSUE THREAD

**Issue by Roger-Z** [milvus-io/milvus#18665]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.0.2
- Deployment mode(standalone or cluster): k8s cluster
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): centos 7
- CPU/Memory: 96GB * 3
- GPU: 
- Others:
```

### Current Behavior

There are 5 querynode in my milvus k8s cluster, however, two pods consume 58GB memory each, other 3 pods only use 3G each. They are not balanced, and I am wondering what will happen when those two pods consume all 96GB memory in the physical machine. 
<img width="430" alt="image" src="https://user-images.githubusercontent.com/3478019/184616949-aef87046-ff58-472d-89e5-4435ee5fd247.png">

### Expected Behavior

memory usage should be balanced between different querynode in a milvus cluster, or there should have an approach to set load balance manually.

**Comment by xiaofan-luan** [comment]
/assign @sunby 
We need to change the balance algorithm soon as possible, can we back port the algorithm of balance by entity numbers to 2.1?

**Comment by Roger-Z** [comment]
Can you plz provide an ETA of fix? our data growing fast, I am really concerned about the consequence when memory usage is larger than the max memory of my server.

**Comment by xiaofan-luan** [comment]
> Can you plz provide an ETA of fix? our data growing fast, I am really concerned about the consequence when memory usage is larger than the max memory of my server.

We have some refactor on 2.2, and it should be able to fix most of the balance issues

**Comment by Folgerjun** [comment]
I have the same problem in milvus 2.1.4.
When I add new querynode, not used.

**Comment by fancy33** [comment]
> > Can you plz provide an ETA of fix? our data growing fast, I am really concerned about the consequence when memory usage is larger than the max memory of my server.
> 
> We have some refactor on 2.2, and it should be able to fix most of the balance issues

@xiaofan-luan Any ETA of this fix 2.2? we also hit same issue, along with our data growing, the memory of one querynode（node1) increased linearly then caused OOMKilled,  but the memory usage of the other querynode(node2) was quite small.
